pkg/helmraiser/helm.go

```diff
@@ -48,114 +69,98 @@ func confToArgs(conf HelmConf) ([]string, []string, error) {
 	// append custom flags to args
 	args = append(args, conf.Flags...)
 
-	if len(args) == 0 {
-		args = nil
-	}
-
 	return args, tempFiles, nil
 }
 
-func parseYamlToMap(yamlFile []byte) (map[string]interface{}, error) {
-	files := make(map[string]interface{})
-	d := yaml.NewDecoder(bytes.NewReader(yamlFile))
-	for {
-		var doc, jsonDoc interface{}
-		if err := d.Decode(&doc); err != nil {
-			if err == io.EOF {
-				break
-			}
-			return nil, errors.Wrap(err, "parsing manifests")
-		}
-
-		jsonRaw, err := json.Marshal(doc)
-		if err != nil {
-			return nil, errors.Wrap(err, "marshaling mainfests")
-		}
+// Template expands a Helm Chart into a regular manifest.List using the `helm
+// template` command
+func (h Helm) Template(name, chart string, opts TemplateOpts) (manifest.List, error) {
+	confArgs, tmpFiles, err := confToArgs(opts)
+	if err != nil {
+		return nil, err
+	}
+	for _, f := range tmpFiles {
+		defer os.Remove(f)
+	}
 
-		if err := json.Unmarshal(jsonRaw, &jsonDoc); err != nil {
-			return nil, errors.Wrap(err, "unmarshaling manifests")
-		}
+	args := []string{name, chart}
+	args = append(args, confArgs...)
 
-		// Unmarshal name and kind
-		kindName := struct {
-			Kind     string `json:"kind"`
-			Metadata struct {
-				Name string `json:"name"`
-			} `json:"metadata"`
-		}{}
-		if err := json.Unmarshal(jsonRaw, &kindName); err != nil {
-			return nil, errors.Wrap(err, "subtracting kind/name through unmarshaling")
-		}
+	cmd := h.run("template", args...)
+	var buf bytes.Buffer
+	cmd.Stdout = &buf
+	cmd.Stderr = os.Stderr
 
-		// snake_case string
-		normalizeName := func(s string) string {
-			s = strings.ReplaceAll(s, "-", "_")
-			s = strings.ReplaceAll(s, ":", "_")
-			s = strings.ToLower(s)
-			return s
-		}
+	if err := cmd.Run(); err != nil {
+		return nil, errors.Wrap(err, "Expanding Helm Chart")
+	}
 
-		// create a map of resources for ease of use in jsonnet
-		name := normalizeName(fmt.Sprintf("%s_%s", kindName.Metadata.Name, kindName.Kind))
-		if jsonDoc != nil {
-			files[name] = jsonDoc
+	var list manifest.List
+	d := yaml.NewDecoder(&buf)
+	for {
+		var m manifest.Manifest
+		if err := d.Decode(&m); err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, errors.Wrap(err, "Parsing Helm output")
 		}
+		list = append(list, m)
 	}
-	return files, nil
+
```

Comment:
If this replaces `parseYamlToMap`, could you move it into a function and add a unit test for it?

Reply:
imo the implementation of yaml unmarshalling doesn't require testing here, because go-yaml is heavily unit tested itself, and unmarshalling yaml output into a `map[string]interface{}` pretty standard Go stuff.

What I instead would like to test here is that the output of `helm` correctly ends up in Jsonnet, regardless of our actual Go implementation.  
I will add such a test in a subsequent pull request, which this one is the preparation for.